sycl/doc/design/OffloadDesign.md

```diff
@@ -0,0 +1,326 @@
+are given further below.
+
+| Option Name                  | Purpose                                      |
+|------------------------------|----------------------------------------------|
+| `--fpga-tool-deps=<arg>`     | Comma separated list of dependency files used for FPGA hardware compiles using `aoc` |
+| `--parallel-link-sycl=<arg>` | Provide the number of parallel jobs that will be used when processing with `llvm-foreach` |
+| `--no-sycl-device-lib=<arg>` | Provide the list of device libraries to restrict from linking during device link |
+
+*Table: Additional Options for clang-linker-wrapper*
+
+The `clang-linker-wrapper` provides an existing option named `-wrapper-jobs`
+that may be useful for our usage instead of creating a new option specific
+to `llvm-foreach` processing.
+
+#### spir64_gen support
+
+Compilation behaviors involving AOT for GPU involve an additional call to
+the OpenCL Offline compiler (OCLOC).  This call occurs after the post-link
+step performed by sycl-post-link.  Additional options passed by the user
+via the `-Xsycl-target-backend=spir64_gen <opts>` command as well as the
+implied options set via target options such as `-fsycl-targets=intel_gpu_skl`
+will be processed by a new options to the wrapper, `--gen-tool-arg=<arg>`
+
+#### spir64_fpga support
+
+Compilation behaviors involving AOT for FPGA involve an additional call to
+the either `aoc` (for Hardware) or `opencl-aot` (for Simulation).  This call
+occurs after the post-link step performed by sycl-post-link.  Additional
+options passed by the user via the `-Xsycl-target-backend=spir64_fpga <opts>`
+command will be processed by a new options to the wrapper,
+`--fpga-tool-arg=<arg>`
+
+The FPGA target also has support for additional generated binaries that
+contain intermediate files specific for FPGA.  These binaries (aoco, aocr and
+aocx) can reside in archives and are treated differently than traditional
+device binaries.
+
+When using the `-fintelfpga` option to enable AOT for FPGA, there are
+additional expectations during the compilation.  Use of the option will enable
+debug generation and also generate dependency information.  The dependency
+generation should be packaged along with the device binary for use during
+the link phase.  If we are not generating an object and performing compilation
+directly to the final executable, a new option named `--fpga-tool-deps=<arg>`
+will be needed to pass along the name of the dependency files created during
+the compile.  The dependency information is only used when compiling for
+hardware.
+
+#### spir64_x86_64 support
+
+Compilation behaviors involving AOT for CPU involve an additional call to
+`opencl-aot`.  This call occurs after the post-link step performed by
+`sycl-post-link`.  Additional options passed by the user via the
+`-Xsycl-target-backend=spir64_x86_64 <opts>` command will be processed by a new
+option to the wrapper, `--cpu-tool-arg=<arg>`
+
+### Integration of the sycl-aspect-filter
```

Comment:
Did you intend to leave this section empty?

FWIW, I think we will change the "optional kernel features" design, so that we do not need the `sycl-aspect-filter` tool.  Instead, I think `sycl-post-link` can do the filtering itself and emit different LLVM bitcode files for each device when in AOT mode.

Maybe this section should just be deleted?

Reply:
It was kind of a placeholder, but I didn't do anything with it.  I will remove.